apisix/discovery/zookeeper/init.lua

```diff
@@ -0,0 +1,209 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local core = require("apisix.core")
+local utils = require("apisix.discovery.zookeeper.utils")
+local schema = require("apisix.discovery.zookeeper.schema")
+local table = require("apisix.core.table")
+local ngx = ngx
+local ipairs = ipairs
+local log = core.log
+
+local _M = {
+    version = 0.1,
+    priority = 1000,
+    name = "zookeeper",
+    schema = schema.schema,
+}
+
+-- Global Configuration
+local local_conf
```

Comment:
It is recommended that the main process request the service instance from ZooKeeper and synchronize it to the worker processes via the shared dict.